lib/sprockets/asset.rb

```diff
@@ -138,7 +138,7 @@ def base64digest
 
     # Public: A "named information" URL for subresource integrity.
     def integrity
-      DigestUtils.integrity_uri(metadata[:digest])
+      DigestUtils.integrity_uri(Digest::SHA256.digest(source))
```

Comment:
Should not the digest already be in the `metadata[:digest]` or in the `digest` method?

Reply:
metadata[:digest] is wrong

Reply:
So it should be fixed. The idea of `metadata[:digest]` is to store the digest of the source. It seems duplication to have another method that does the same thing.

Reply:
I think this change is the reason https://github.com/rails/sprockets/commit/64fadf90882761ed5baac3546c4435c600c50ae5.

We added the version to the digest but that is wrong in my opinion. We should add to the cache key, not to the digest itself.

Reply:
yes, this change

Reply:
There are two paths:

1. We revert that change and add a test that fails when it is added
2. We change the code to make version to change the cache key but not the digest and we add a test that fails when the version is added to the digest.

Reply:
2. What's key should be added? `metadata[:version]`?

Reply:
Yes, that could be a good name for the key, but I guess we don't need to add this to the metadata given the environment already have the version.